.github/workflows/cmake/dependencies_nvcc.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+#
+# Copyright 2020 Axel Huebl
+#
+# License: BSD-3-Clause-LBNL
+
+# search recursive inside a folder if a file contains tabs
+#
+# @result 0 if no files are found, else 1
+#
+
+set -eu -o pipefail
+
+sudo apt-get install -y --no-install-recommends\
+    build-essential     \
+    cmake               \
+    g++-5               \
+    gfortran-5          \
+    libopenmpi-dev      \
+    openmpi-bin         \
+    nvidia-cuda-dev     \
+    nvidia-cuda-toolkit
+
+# Patch broken GCC 5.5 libs in <algorithm>
+#   https://gcc.gnu.org/bugzilla/show_bug.cgi?id=76731
+#   https://stackoverflow.com/a/50815334/2719194
+for f in avx512fintrin.h avx512pfintrin.h avx512vlintrin.h
```

Comment:
I wish this wasn't necessary, but:
- CUDA 9.1 was incompatible with `std::tuple` in GCC 6
- GCC 5.5 ships broken stdlib headers with the issues linked herein

Reply:
Upstream bug report with Ubuntu filed as: https://bugs.launchpad.net/ubuntu/+source/nvidia-cuda-toolkit/+bug/1865091